src/lib/registration/connect_helpers.rb

```diff
@@ -139,7 +146,8 @@ def self.catch_registration_errors(message_prefix: "",
         # update the message when an old SMT server is found
         check_smt_api(e.message)
 
-        report_error(message_prefix + _("Connection to registration server failed."), e.message)
+        report_error(message_prefix + _("Connection to registration server failed."),
```

Comment:
The "Connection to registration server failed" error is actually a bit misleading in this case. The connection was OK, but the data could not be parsed. So I'd replace the original message completely. (IIRC I used this message because of the text freeze at that time.)

And I'm not sure if we should mention JSON at all, this is an implementation detail and it cannot help the user to fix the problem in any way. And as we use the SUSEConnect libarry the underlying protocol theoretically might changed without YaST noticing any difference.

```ruby
report_error(message_prefix + _("Cannot parse the data from server."), e.message)
```